library/entity/multipart/src/main/java/org/quiltmc/qsl/multipart/mixin/EnderDragonEntityMixin.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.multipart.mixin;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyArg;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.boss.dragon.EnderDragonEntity;
+import net.minecraft.entity.boss.dragon.EnderDragonPart;
+
+import org.quiltmc.qsl.multipart.api.EntityPart;
+import org.quiltmc.qsl.multipart.api.MultipartEntity;
+
+@Mixin(EnderDragonEntity.class)
+public abstract class EnderDragonEntityMixin implements MultipartEntity {
+	@Shadow
+	public abstract EnderDragonPart[] getBodyParts();
+
+	@Override
+	public EntityPart<?>[] getEntityParts() {
+		return this.getBodyParts();
+	}
+
+	/**
+	 * <p>The vanilla method sets the {@link EnderDragonEntity#head} to
+	 * the same {@link Entity#getId() id} as the {@link EnderDragonEntity} herself.</p>
+	 */
+	@ModifyArg(method = "onSpawnPacket", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/boss/dragon/EnderDragonPart;setId(I)V"))
+	private int setProperId(int oldId) {
+		return oldId + 1;
+	}
```

Comment:
This would be a desync issue when you have a Quilt client (with this library installed of course) connecting to a vanilla-compatible server

Reply:
I'm just in a silly goofy mood. The vanilla server expects this behavior, it's the vanilla client that has a bug. See <https://bugs.mojang.com/browse/MC-225055>. That bug is due to entity id desyncs.

Reply:
Technically, that makes qsl a cheat mod because now you have an unfair advantage over vanilla clients. Hmmm